sprintf is deprecated in Mac OSX SDK

```
In file included from ../src/numerics/laspack_vector.C:27:
./include/libmesh/laspack_vector.h:332:8: error: 'sprintf' is deprecated: This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead. [-Werror,-Wdeprecated-declarations]
  std::sprintf(foo,  "Vec-%d", cnt++);
       ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdio.h:188:1: note: 'sprintf' has been explicitly marked deprecated here
__deprecated_msg("This function is provided for compatibility reasons only.  Due to security concerns inherent in the design of sprintf(3), it is highly recommended that you use snprintf(3) instead.")
^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:215:48: note: expanded from macro '__deprecated_msg'
        #define __deprecated_msg(_msg) __attribute__((__deprecated__(_msg)))
```

A cursory search unveiled a few other instances as well:
```
include/numerics/laspack_vector.h:  std::sprintf(foo,  "Vec-%d", cnt++);
src/systems/frequency_system.C:  sprintf(buf, "frequency %04u", n);
src/systems/frequency_system.C:  sprintf(buf, "solution %04u", n);
src/systems/system_io.C:#include <cstdio> // for std::sprintf
src/systems/system_io.C:        std::sprintf(buf, "%u", var);
src/systems/system_io.C:        std::sprintf(buf, "%s", this->variable_name(var).c_str());
src/systems/system_io.C:        std::sprintf(buf, "%s", this->variable_name(var).c_str());
src/systems/system_io.C:        std::sprintf(buf, "%s", this->variable_name(var).c_str());
src/systems/system_io.C:        std::sprintf(buf, "%s", this->variable_name(var).c_str());
src/systems/system_io.C:        std::sprintf(buf, "%s", this->variable_name(var).c_str());
src/systems/system_io.C:        std::sprintf(buf, "%s", this->variable_name(var).c_str());
src/systems/system_io.C:            std::sprintf(buf, "%dth vector", cnt++);
```
so we should probably get rid of these ASAP.

See also: https://civet.inl.gov/job/1226629/

--- a/include/numerics/laspack_vector.h
+++ b/include/numerics/laspack_vector.h
@@ -35,7 +35,6 @@
 #include <qvector.h>
 
 // C++ includes
-#include <cstdio> // for std::sprintf
 #include <limits>
 #include <mutex>
 
@@ -328,10 +327,10 @@ void LaspackVector<T>::init (const numeric_index_type n,
   // create a sequential vector
 
   static int cnt = 0;
-  char foo[80];
-  std::sprintf(foo,  "Vec-%d", cnt++);
+  std::string foo{"Vec-"};
+  foo += std::to_string(cnt++);
 
-  V_Constr(&_vec, const_cast<char *>(foo), n, Normal, _LPTrue);
+  V_Constr(&_vec, const_cast<char *>(foo.c_str()), n, Normal, _LPTrue);
 
   this->_is_initialized = true;
 #ifndef NDEBUG

--- a/src/systems/frequency_system.C
+++ b/src/systems/frequency_system.C
@@ -32,9 +32,6 @@
 #include "libmesh/linear_solver.h"
 #include "libmesh/numeric_vector.h"
 
-// C++ includes
-#include <cstdio>          // for sprintf
-
 namespace libMesh
 {
 
@@ -402,9 +399,11 @@ void FrequencySystem::set_current_frequency(unsigned int n)
 std::string FrequencySystem::form_freq_param_name(const unsigned int n) const
 {
   libmesh_assert_less (n, 9999);
-  char buf[15];
-  sprintf(buf, "frequency %04u", n);
-  return (buf);
+  std::string nstr = std::to_string(n);
+  constexpr std::size_t digits = 4;
+  const int zeros = digits - std::min(digits,nstr.size());
+  nstr.insert(0, zeros, '0');
+  return "frequency "+nstr;
 }
 
 
@@ -412,9 +411,11 @@ std::string FrequencySystem::form_freq_param_name(const unsigned int n) const
 std::string FrequencySystem::form_solu_vec_name(const unsigned int n) const
 {
   libmesh_assert_less (n, 9999);
-  char buf[15];
-  sprintf(buf, "solution %04u", n);
-  return (buf);
+  std::string nstr = std::to_string(n);
+  constexpr std::size_t digits = 4;
+  const int zeros = digits - std::min(digits,nstr.size());
+  nstr.insert(0, zeros, '0');
+  return "solution "+nstr;
 }
 
 } // namespace libMesh

--- a/src/systems/system_io.C
+++ b/src/systems/system_io.C
@@ -31,7 +31,6 @@
 
 
 // C++ Includes
-#include <cstdio> // for std::sprintf
 #include <memory>
 #include <numeric> // for std::partial_sum
 #include <set>
@@ -1311,7 +1310,6 @@ void System::write_header (Xdr & io,
     return;
 
   std::string comment;
-  char buf[80];
 
   // 5.)
   // Write the number of variables in the system
@@ -1334,8 +1332,7 @@ void System::write_header (Xdr & io,
       {
         // set up the comment
         comment  = "#   Name, Variable No. ";
-        std::sprintf(buf, "%u", var);
-        comment += buf;
+        comment += std::to_string(var);
         comment += ", System \"";
         comment += this->name();
         comment += "\"";
@@ -1348,8 +1345,7 @@ void System::write_header (Xdr & io,
       {
         // set up the comment
         comment  = "#     Subdomains, Variable \"";
-        std::sprintf(buf, "%s", this->variable_name(var).c_str());
-        comment += buf;
+        comment += this->variable_name(var);
         comment += "\", System \"";
         comment += this->name();
         comment += "\"";
@@ -1366,8 +1362,7 @@ void System::write_header (Xdr & io,
       {
         // set up the comment
         comment = "#     Approximation Order, Variable \"";
-        std::sprintf(buf, "%s", this->variable_name(var).c_str());
-        comment += buf;
+        comment += this->variable_name(var);
         comment += "\", System \"";
         comment += this->name();
         comment += "\"";
@@ -1382,8 +1377,7 @@ void System::write_header (Xdr & io,
       // do the same for radial_order
       {
         comment = "#     Radial Approximation Order, Variable \"";
-        std::sprintf(buf, "%s", this->variable_name(var).c_str());
-        comment += buf;
+        comment += this->variable_name(var);
         comment += "\", System \"";
         comment += this->name();
         comment += "\"";
@@ -1399,8 +1393,7 @@ void System::write_header (Xdr & io,
       {
         // set up the comment
         comment = "#     FE Family, Variable \"";
-        std::sprintf(buf, "%s", this->variable_name(var).c_str());
-        comment += buf;
+        comment += this->variable_name(var);
         comment += "\", System \"";
         comment += this->name();
         comment += "\"";
@@ -1412,8 +1405,7 @@ void System::write_header (Xdr & io,
 #ifdef LIBMESH_ENABLE_INFINITE_ELEMENTS
 
         comment = "#     Radial FE Family, Variable \"";
-        std::sprintf(buf, "%s", this->variable_name(var).c_str());
-        comment += buf;
+        comment += this->variable_name(var);
         comment += "\", System \"";
         comment += this->name();
         comment += "\"";
@@ -1422,8 +1414,7 @@ void System::write_header (Xdr & io,
         io.data (radial_fam, comment.c_str());
 
         comment = "#     Infinite Mapping Type, Variable \"";
-        std::sprintf(buf, "%s", this->variable_name(var).c_str());
-        comment += buf;
+        comment += this->variable_name(var);
         comment += "\", System \"";
         comment += this->name();
         comment += "\"";
@@ -1456,19 +1447,16 @@ void System::write_header (Xdr & io,
           {
             // 9.)
             // write the name of the cnt-th additional vector
-            comment =  "# Name of ";
-            std::sprintf(buf, "%dth vector", cnt++);
-            comment += buf;
+            const std::string dth_vector = std::to_string(cnt++)+"th vector";
+            comment =  "# Name of " + dth_vector;
             std::string vec_name = pr.first;
 
             io.data (vec_name, comment.c_str());
             int vec_projection = _vector_projections.at(vec_name);
-            comment = "# Whether to do projections for ";
-            comment += buf;
+            comment = "# Whether to do projections for " + dth_vector;
             io.data (vec_projection, comment.c_str());
             int vec_type = _vector_types.at(vec_name);
-            comment = "# Parallel type of ";
-            comment += buf;
+            comment = "# Parallel type of " + dth_vector;
             io.data (vec_type, comment.c_str());
           }
       }

